### PR TITLE
Merge parameters parsing and match in a single method

### PR DESCRIPTION
**Pull request by romainpiel**
See #66 for more details. I'm not sure about the inclusion of `Optional<T>` but I included a very light version of it. We can definitely get rid of it too.

### CODE REVIEW

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkMatch.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  public DeepLinkEntry getEntry() {
+    return entry;
+  }
+
+  public Map<String, String> getParameters() {
+    return parameters;
+  }
+
+  public DeepLinkUri getUri() {
```
Comment (felipecsl) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkMatch.java]: same for these 3 methods

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/Optional.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.airbnb.deeplinkdispatch;
+
+public class Optional<T> {
```
Comment (felipecsl) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/Optional.java]: package private, 2 spaces indentation

File: deeplinkdispatch/src/test/java/com/airbnb/deeplinkdispatch/DeepLinkEntryTest.java [hunk 3]
```diff
@@ -11,13 +11,20 @@
   @Test public void testSingleParam() {
     DeepLinkEntry entry = deepLinkEntry("airbnb://foo/{bar}");
 
-    assertThat(entry.getParameters("airbnb://foo/myData").get("bar")).isEqualTo("myData");
+    final Optional<DeepLinkMatch> matches = entry.matches("airbnb://foo/myData");
```
Comment (felipecsl) [inline deeplinkdispatch/src/test/java/com/airbnb/deeplinkdispatch/DeepLinkEntryTest.java]: nit: you don't need to make these local variables final, can remove to reduce clutter

**Comment by felipecsl** [comment]
Looking pretty good overall. I'm not sure why Travis did not run for this PR though. I'll look into that

**Comment by felipecsl** [comment]
According to [this](https://docs.travis-ci.com/user/pull-requests) it could be because your branch is outdated. Can you try rebasing it against master, please?

**Comment by RomainPiel** [comment]
Sorry for the noise, I can squash the commits if you prefer

**Comment by rossbacher** [comment]
Not applicable anymore with new matcher.

**Pull request closed without merging**